vortex-layout/src/vtable.rs

```diff
@@ -1,60 +1,125 @@
 use std::collections::BTreeSet;
-use std::fmt::{Debug, Display, Formatter};
+use std::fmt::Debug;
+use std::ops::Deref;
 use std::sync::Arc;
 
-use arcref::ArcRef;
-use vortex_array::ArrayContext;
-use vortex_dtype::FieldMask;
+use vortex_array::{ArrayContext, DeserializeMetadata, SerializeMetadata};
+use vortex_dtype::{DType, FieldMask};
 use vortex_error::VortexResult;
 
-use crate::segments::SegmentSource;
-use crate::{Layout, LayoutId, LayoutReader};
+use crate::children::LayoutChildren;
+use crate::segments::{SegmentId, SegmentSource};
+use crate::{
+    IntoLayout, Layout, LayoutChildType, LayoutEncoding, LayoutEncodingRef, LayoutId,
+    LayoutReaderRef, LayoutRef,
+};
 
-/// A reference to a layout VTable, either static or arc'd.
-pub type LayoutVTableRef = ArcRef<dyn LayoutVTable>;
+pub trait VTable: 'static + Sized + Send + Sync + Debug {
+    type Layout: 'static + Send + Sync + Clone + Deref<Target = dyn Layout> + IntoLayout;
```

Comment:
If we want to have Debug for Layouts we should add Debug here so it correctly recurses?